presto-orc/src/main/java/com/facebook/presto/orc/reader/ApacheHiveTimestampDecoder.java

```diff
@@ -22,7 +22,8 @@ private ApacheHiveTimestampDecoder() {}
     // This comes from the Apache Hive ORC code
     public static long decodeTimestamp(long seconds, long serializedNanos, DecodeTimestampOptions options)
     {
-        long value = (seconds + options.getBaseSeconds()) * options.getUnitsPerSecond();
+        long secondsWithBase = seconds + options.getBaseSeconds();
```

Comment:
Wondering if this line can this create an overflow (Is `addExact` needed)

Reply:
I don't think this is needed here. Long variable is used to represent timestamp with milliseconds generally (in java.sql.Timestamp, etc). We're only handling seconds in long variable here.

Although it is needed where nanos is truncated and added to value. I have included this change now.